### PR TITLE
Issue6

### LINKED ISSUE #6: [backend] Wire GetGearList handler to dynamodb
## Background
The GetGearList lambda handler currently delegates fetching the list of gear to the `GearProvider`.  The `GearProvider` class has a hardcoded gear model that it returns, regardless of input (ignores incoming request).

## TODO
* Create AWS dagger module 
  * Sets up credential provider
  * Sets up dynamodb client (and probably mapper)
* Refactor `GearProvider` to be an interface with an implementation
* Have the GearProvider impl use the dynamodb client to fetch gear
* Write unit tests for the `GetGearList` and `GearProvider` (implementation) classes

> Note: this task does not involve updating the cloudformation template to set up the dynamodb table/resources.

> Note: this task does not involve updating the `GetGearList` handler to use the incoming request to extract the profileId to fetch the gear list for.  We'll just use a hardcoded id (i.e. id=1) for now.

### PR DESCRIPTION
**Pull request by nathanbaney**
Did issue 6 stuff

### CODE REVIEW

File: caampr-backend/build.gradle [hunk 1]
```diff
@@ -22,9 +22,15 @@ targetCompatibility = 1.8
 repositories {
     // Use JCenter for resolving dependencies.
     jcenter()
+    mavenCentral()
 }
 
 dependencies {
+    implementation platform('com.amazonaws:aws-java-sdk-bom:1.11.969')
```
Comment (jebaney) [inline caampr-backend/build.gradle:29]: what is this `platform` doing? why can't we just bring in the the dynamodb client/model dependencies in the implementation section, like I did with the lambda stuff?

File: caampr-backend/build.gradle [hunk 2]
```diff
@@ -22,9 +22,15 @@ targetCompatibility = 1.8
...
+    implementation (            
+        'com.amazonaws:aws-java-sdk-s3',
```
Comment (jebaney) [inline caampr-backend/build.gradle]: do we need s3 here?

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Link.java [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 @Data
+@NoArgsConstructor
+@AllArgsConstructor
 public class Link {
-    private final int id;
-    private final String url;
-    private final LinkTypes type;
+    private int id;
+    private String url;
+    private LinkTypes type;
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Link.java:15]: note that this is probably gonna fail without the proper enum annotation. can be fixed in another commit

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.jelistan.caampr.lambda.provider;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.jelistan.caampr.lambda.model.Gear;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class GearListProvider implements GearProvider{
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java]: `GearListProvider` may be a bit too narrow here.  I think this should probably just be `GearProviderImpl` or `DynamnoGearProvider` (dynamo implementation of the gear provider interface) as it should provide methods to interact with all facets of gear (fetch single, create, fetch list, etc)

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    private final DynamoDBMapper mapper;
+
+    @Inject
+    public GearListProvider(DynamoDBMapper mapper){
+        this.mapper = mapper;
+    }
+
+    public List<Gear> getGearList(final String profileId){
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java]: public methods should have valid javadocs

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearProvider.java [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
 import com.jelistan.caampr.lambda.model.Gear;
 import com.jelistan.caampr.lambda.model.Link;
 import com.jelistan.caampr.lambda.model.LinkTypes;
 
 import java.util.List;
 
-public class GearProvider {
+public interface GearProvider {
 
     public static final Gear FAKE_GEAR = Gear.builder()
```
Comment (jebaney) [inline caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearProvider.java]: a constant defined in an interface does not require the `public static final` part, it's redundant

**Review by jebaney — CHANGES_REQUESTED** [review]

File: caampr-backend/build.gradle (hunk 1 above)
Comment (nathanbaney) [reply caampr-backend/build.gradle:29]: required for gradle build, from what i can gather from docs

File: caampr-backend/build.gradle (hunk 2 above)
Comment (nathanbaney) [reply caampr-backend/build.gradle]: removed s3 dep

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Link.java (hunk 3 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/model/Link.java:15]: todo, in issue 7

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java (hunk 4 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java]: renamed, added getgear method that takes gearId as param

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearProvider.java (hunk 6 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearProvider.java]: removed final

File: caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java (hunk 5 above)
Comment (nathanbaney) [reply caampr-backend/src/main/java/com/jelistan/caampr/lambda/provider/GearListProvider.java]: added docs for methods

**Pull request merged**